cpu/nrf52/periph/i2c.c

```diff
@@ -45,17 +45,25 @@
  */
 static mutex_t locks[I2C_NUMOF];
 
+/**
+ * @brief   array with a busy mutex for each I2C device, used to block the
+ *          thread until the transfer is done
+ */
+static mutex_t busy[I2C_NUMOF];
+
+void i2c_isr_handler(void *arg);
+
 static inline NRF_TWIM_Type *bus(i2c_t dev)
 {
     return i2c_config[dev].dev;
 }
 
 static int finish(i2c_t dev)
 {
+    /* Unmask interrupts */
+    bus(dev)->INTENSET = TWIM_INTEN_STOPPED_Msk | TWIM_INTEN_ERROR_Msk;
+    mutex_lock(&busy[dev]);
     DEBUG("[i2c] waiting for STOPPED or ERROR event\n");
```

Comment:
Should be moved before the lock - just squash again

Reply:
Fixed!